basic/token-burn/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use scrypto::prelude::*;
+
+blueprint! {
+
+    struct TokenBurn {
+        token_address: Address,
+        burnt_amount: Decimal,
+    }
+
+    impl TokenBurn {
+        pub fn new(token_address: Address) -> Component {
+            Self {
+                token_address: token_address,
+                burnt_amount: Decimal::zero(),
+            }
+            .instantiate()
+        }
+
+        pub fn burn_tokens(&mut self, tokens_to_burn: Bucket) {
+            self.burnt_amount += tokens_to_burn.amount();
+            Vault::new(ResourceDef::from(self.token_address)).put(tokens_to_burn);
```

Comment:
I like the idea of allowing people to "effectively" burn any tokens. Although, the current implementation may not work once dangling vault check is enforced. After that, will always need to attach the vaults to a component.